src/context.rs

```diff
@@ -105,10 +106,83 @@ fn test_should_add_a_service() {
         }
     }
 
-    let service = ServiceStub;
-    let mut foo = Context::new(false, Some("localhost".to_owned()), None, None);
+    mod add_service {
+        use super::super::*;
 
-    assert_eq!(foo.services.is_empty(), true);
-    foo.add_service(Box::new(service));
-    assert_eq!(foo.services.is_empty(), false);
+        #[test]
+        fn test_should_increase_number_of_services() {
+            let service = super::ServiceStub;
+            let context = Context::shared(false, Some("localhost".to_owned()), None, None);
+            let mut locked_context = context.lock().unwrap();
```

Comment:
In a perfect world, these variables should be initialized in a setup-like function. I tried to play with that. I have not managed to return the correct type, so the compiler wasn't very happy.

Also, if we declared the variables in the module itself, they should be `static mut`. Using a `static mut` variable needs to be done in an `unsafe` block. I don't think this is a good practice, even for testing.